lib/charms/kubernetes_charm_libraries/v0/kubernetes_volumes_patch.py

```diff
@@ -0,0 +1,531 @@
+                    f"Could not get statefulset {statefulset_name}"
+                )
+            return False
+        return self._statefulset_volumes_are_patched(
+            statefulset_spec=statefulset.spec,  # type: ignore[attr-defined]
+            requested_volumes=requested_volumes,
+        )
+
+    @staticmethod
+    def _statefulset_volumes_are_patched(
+        statefulset_spec: StatefulSetSpec,
+        requested_volumes: Iterable[RequestedVolume],
+    ) -> bool:
+        """Returns whether a StatefulSet is patched with requested volumes.
+
+        Args:
+            statefulset_spec: StatefulSet spec
+            requested_volumes: Iterable of requested volumes
+
+        Returns:
+            bool
+        """
+        if not statefulset_spec.template.spec.volumes:
+            return False
+        return all(
+            [
+                requested_volume.volume in statefulset_spec.template.spec.volumes
+                for requested_volume in requested_volumes
+            ]
+        )
+
+    def _pod_volumemounts_contain_requested_volumes(
+        self,
+        containers: Iterable[Container],
+        container_name: str,
+        requested_volumes: Iterable[RequestedVolume],
+    ) -> bool:
+        """Returns whether container spec contains the expected requested volumes mounts.
+
+        Args:
+            containers: Iterable of Containers
+            container_name: Container name
+            requested_volumes: Requested volumes we expect to be set
+
+        Returns:
+            bool
+        """
+        container = self._get_container(container_name=container_name, containers=containers)
+        return all(
+            [
+                requested_volume.volume_mount in container.volumeMounts
+                for requested_volume in requested_volumes
+            ]
+        )
+
+    def _pod_resources_are_set(
+        self,
+        containers: Iterable[Container],
+        container_name: str,
+        requested_volumes: Iterable[RequestedVolume],
+    ) -> bool:
+        """Returns whether container spec contains the expected resources requests and limits.
+
+        Args:
+            containers: Iterable of Containers
+            container_name: Container name
+            requested_volumes: Iterable of requested volumes
+
+        Returns:
+            bool
+        """
+        container = self._get_container(container_name=container_name, containers=containers)
+        if not self._hugepages_in_requested_volumes(requested_volumes):
+            return True
+        if not container.resources.limits or not container.resources.requests:
+            return False
+        for limit, value in HUGEPAGES_RESOURCES_LIMITS.items():
+            if container.resources.limits.get(limit) != value:
+                return False
+        for request, value in HUGEPAGES_RESOURCES_REQUESTS.items():
+            if container.resources.requests.get(request) != value:
+                return False
```

Comment:
Just nitpicking, but IMO it would look cleaner if we had 2 separate methods here - `_hugepages_resources_limits_set` and `_hugepages_resources_requests_set`.
WDYT?